### PR TITLE
several workflow features and a critical fix to storm syntax and prop normalization

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/tests/test_cortex.py [hunk 1]
```diff
@@ -1496,3 +1496,8 @@ def test_cortex_syncs_errs(self):
             self.eq( len(errs), 1 )
             self.eq( errs[0][0][1]['mesg'][0], 'newp:fake' )
             self.nn( core.getTufoByProp('inet:fqdn','vertex.link') )
+
+    def test_cortex_norm_fail(self):
+        with s_cortex.openurl('ram:///') as core:
+            core.formTufoByProp('inet:netuser','vertex.link/visi')
+            self.raises( BadTypeValu, core.eval, 'inet:netuser="totally invalid input"' )
```
Comment (vEpiphyte) [inline synapse/tests/test_cortex.py]: :+1:

File: synapse/cmds/cortex.py [hunk 2]
```diff
@@ -39,33 +45,52 @@ def runCmdOpts(self, opts):
                 took = opfo.get('took')
                 self.printf('    %s (took:%d) %r' % (mnem,took,opfo))
 
+            self.printf('')
```
Comment (vEpiphyte) [inline synapse/cmds/cortex.py]: Extraneous printf() call.

File: synapse/cmds/cortex.py [hunk 3]
```diff
@@ -39,33 +45,52 @@ def runCmdOpts(self, opts):
...
+
             self.printf('options:')
             for name,valu in sorted(resp.get('options').items()):
                 self.printf('    %s = %s' % (name,valu))
 
+            self.printf('')
```
Comment (vEpiphyte) [inline synapse/cmds/cortex.py]: Extraneous printf() call.

File: synapse/cmds/cortex.py [hunk 4]
```diff
@@ -39,33 +45,52 @@ def runCmdOpts(self, opts):
...
+
             self.printf('limits:')
             for name,valu in sorted(resp.get('limits').items()):
                 self.printf('    %s = %s' % (name,valu))
 
+            self.printf('')
```
Comment (vEpiphyte) [inline synapse/cmds/cortex.py]: Extraneous printf() call.

File: synapse/cores/ram.py [hunk 5]
```diff
@@ -54,13 +54,20 @@ def _tufosByLe(self, prop, valu, limit=None):
         return self.getTufosByIdens([ r[0] for r in rows ])
 
     def _sizeByRange(self, prop, valu, limit=None):
```
Comment (vEpiphyte) [inline synapse/cores/ram.py]: limit is unused in this function

File: synapse/lib/storm.py [hunk 6]
```diff
@@ -14,6 +15,25 @@
 
 logger = logging.getLogger(__name__)
 
+class LimitHelp:
+
+    def __init__(self, limit):
+        self.limit = limit
```
Comment (vEpiphyte) [inline synapse/lib/storm.py]: ```int(limit)``` So we blow up right away on bad limit values.
also blow up on limit's < 0

File: synapse/models/inet.py [hunk 7]
```diff
@@ -255,6 +256,25 @@ def getDataModel():
 
             ]),
 
+            ('inet:netfile',{},[
```
Comment (vEpiphyte) [inline synapse/models/inet.py]: Do want want to add a netpost or url to this as a secondary property?

File: synapse/models/orgs.py [hunk 8]
```diff
@@ -14,6 +14,8 @@ def getDataModel():
             ('ou:sic',{'subof':'int','doc':'Standard Industrial Classification Code'}),
             ('ou:naics',{'subof':'int','doc':'North American Industry Classification System'}),
 
+            ('ou:suborg', {'subof':'comp','fields':'org,ou:org|sub,ou:org','doc':'An org which owns a sub org'}),
```
Comment (vEpiphyte) [inline synapse/models/orgs.py]: Missing test for suborg creation

File: synapse/models/inet.py [hunk 9]
```diff
@@ -53,6 +53,7 @@ def getDataModel():
             ('inet:netgroup',   {'subof':'sepr','sep':'/','fields':'site,inet:fqdn|name,ou:name','doc':'A group within an online community'}),
 
             ('inet:netpost',    {'subof':'comp','fields':'netuser,inet:netuser|text,str:txt', 'doc':'A post made by a netuser'}),
+            ('inet:netfile',    {'subof':'comp','fields':'netuser,inet:netuser|file,file:bytes', 'doc':'A file posted by a netuser'}),
```
Comment (vEpiphyte) [inline synapse/models/inet.py]: missing test for netfile creation

File: synapse/lib/storm.py [hunk 10]
```diff
@@ -216,11 +238,15 @@ def __init__(self, **opts):
...
-        self.setOperFunc('deltag',self._stormOperDelTag)
-        self.setOperFunc('nexttag',self._stormOperNextSeq)
-        self.setOperFunc('setprop',self._stormOperSetProp)
+        self.setOperFunc('alltag', self._stormOperAllTag)
+        self.setOperFunc('addtag', self._stormOperAddTag)
+        self.setOperFunc('deltag', self._stormOperDelTag)
+        self.setOperFunc('nexttag', self._stormOperNextSeq)
+        self.setOperFunc('setprop', self._stormOperSetProp)
+
+        self.setOperFunc('addxref', self._stormOperAddXref)
```
Comment (vEpiphyte) [inline synapse/lib/storm.py]: fyi - this operator is uedntest

File: synapse/lib/storm.py (hunk 6 above)
Comment (invisig0th) [inline synapse/lib/storm.py]: it needs to be ok for limit to be None ( our helper makes the user not need to worry about limit=None case )

File: synapse/models/inet.py (hunk 7 above)
Comment (invisig0th) [inline synapse/models/inet.py]: Possibly.  If so, lets noodle it and do that in another delta ( adding a new secondary prop is pretty harmless )

**Comment by vEpiphyte** [comment]
Approving per discussion and backlogging some items.

**Pull request merged**